lib/iopromise/data_loader.rb

```diff
@@ -4,11 +4,30 @@ module IOPromise
   module DataLoader
     module ClassMethods
       def attr_promised_data(*args)
+        build_func = if args.last&.is_a?(Proc)
+          args.pop
+        else
+          nil
+        end
+
         @promised_data_keys ||= []
         @promised_data_keys.concat(args)
 
         args.each do |arg|
```

Comment:
Any thoughts on taking this opportunity to only allow one attribute per `attr_promised_data` call? I am not convinced the ability to have a single block apply to multiple arg attributes is all that useful, but maybe I'm not seeing something.

Reply:
Oh, I guess my comment there was really for this point - but passing the `arg` was just an extension of matching `attr_reader` initially. As per my previous comment, let's simplify all this for now :+1: